spec/crypto_env_var_spec.rb

```diff
@@ -1,6 +1,201 @@
 require "spec_helper"
 
 RSpec.describe CryptoEnvVar do
+  describe "bootstrapping the ENV Hash" do
+    def clean_the_env
+      ENV.delete(CryptoEnvVar::CRYPTO_ENV_VAR)
+      ENV.delete(CryptoEnvVar::DECRYPT_KEY_VAR)
+      SECRET_HASH.keys.each { |k| ENV.delete(k) }
+    end
+
+    before { clean_the_env }
+    after(:all) { clean_the_env }
+
+    # make this available in the class scope
+    SECRET_HASH = {
+      "ROMEO" => "Romeo and Juliet (1976)",
+      "IAGO" => "Othello (1989)",
+      "MAGNETO" => "X-Men (2000)",
+      "GANDALF" => "The Fellowship of the Ring (2001)",
+    }
+
+    let(:secret_hash) { SECRET_HASH }
+    let(:encrypted_env) do
+      described_class.encrypt(secret_hash, private_key)
+    end
+
+    shared_examples_for "it populates the ENV with the encrypted variables" do |options|
+      to_skip = Array((options || {})[:except])
+      to_test = SECRET_HASH.reject { |k, _| to_skip.include?(k) }
+
+      to_test.each do |k, v|
+        it "sets the #{k} key" do
+          expect { subject }.to change { ENV[k] }.from(nil).to(v)
+        end
+      end
+    end
+
+    shared_examples_for "it does NOT populate the ENV with the encrypted variables" do
+      SECRET_HASH.each do |k, v|
+        it "doesn't set the #{k} key" do
+          expect { subject rescue nil }.to_not change { ENV[k] }
+        end
+      end
+    end
+
+    shared_examples_for "empty source ENV, bootstrapping failures" do
+      describe "when the ENV does NOT contain the default source data" do
+        context "any of them" do
+          it "raises a KeyError error" do
+            expect { subject }.to raise_error(KeyError)
+          end
+
+          it_behaves_like "it does NOT populate the ENV with the encrypted variables"
+        end
+
+        context "no CRYPTO_ENV source ENV var" do
+          before do
+            ENV[CryptoEnvVar::DECRYPT_KEY_VAR] = public_key
+          end
+
+          it "raises a KeyError error" do
+            expect { subject }.to raise_error(KeyError)
+          end
+
+          it_behaves_like "it does NOT populate the ENV with the encrypted variables"
+        end
+
+        context "no DECRYPT_KEY source ENV var" do
+          before do
+            ENV[CryptoEnvVar::CRYPTO_ENV_VAR] = encrypted_env
```

Comment:
If you allowed replacing `ENV` with an arbitrary hash in the function signature, you wouldn't need that.

Reply:
This is actually to control what's read from the default lambda used to read the encrypted env. As I explained in the other comments, injecting the `env` in that introduces some other API issues. I'd be happy to hear about alternative solutions though.

Reply:
What I'd personally do in such case is make those callables "partially applied" by using plain objects:

```rb
class Exporter
  def self.export(to: ENV)
    new(target: to).export
  end
end
```

But look, this is an academic discussion, feel free to 🛳 it.

Reply:
I see what you mean. 👍  Thanks.

Possibly something for another PR, so we can use this in the meantime.